jicofo/src/main/java/org/jitsi/jicofo/conference/JitsiMeetConferenceImpl.java

```diff
@@ -1519,6 +1553,60 @@ private void rescheduleSingleParticipantTimeout()
         logger.info("Scheduled single person timeout.");
     }
 
+    /** Called when a new visitor has been added to the conference. */
+    private void visitorAdded()
+    {
+        updateVisitors(+1);
+    }
+
+    /** Called when a new visitor has been added to the conference. */
+    private void visitorRemoved()
+    {
+        updateVisitors(-1);
+    }
+
+    /** Called when the number of visitors in the conference has changed. */
+    private void updateVisitors(int delta)
+    {
+        synchronized(this)
+        {
+            numVisitors += delta;
+            if (numVisitors < 0) {
+                /* Something's gone wrong with our counting */
+                logger.error("numVisitors is " + numVisitors + ", something's wrong");
+            }
+            Instant now = Instant.now();
+            if (sendNumVisitorsTask != null)
+            {
+                return;
+            }
+            if (visitorsLastUpdated != null &&
+                Duration.between(visitorsLastUpdated, now).
+                    compareTo(VisitorsConfig.config.getNotificationInterval()) < 0)
+            {
+                sendNumVisitorsTask = TaskPools.getIoPool().submit(this::sendNumVisitors);
```

Comment:
Does this need to be scheduled for later?

Reply:
🤦